Source/Core/Core/PowerPC/Interpreter/Interpreter_Tables.cpp

```diff
@@ -5,492 +5,492 @@
+    table[tpl.opcode] = tpl.fn;
+  };
 
-  for (int i = 0; i < 32; i++)
+  for (u32 i = 0; i < 32; i++)
   {
-    int fill = i << 5;
-    for (auto& tpl : table63_2)
+    const u32 fill = i << 5;
+    for (const auto& tpl : s_table63_2)
     {
-      int op = fill + tpl.opcode;
-      m_op_table63[op] = tpl.Inst;
-      m_infoTable63[op] = &tpl.opinfo;
+      const u32 op = fill + tpl.opcode;
+      ASSERT(table[op] == Interpreter::unknown_instruction);
+      table[op] = tpl.fn;
     }
   }
 
-  m_numInstructions = 0;
-  for (auto& tpl : primarytable)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table4_2)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table4_3)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table4)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table31)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table19)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table59)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table63)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-  for (auto& tpl : table63_2)
-    m_allInstructions[m_numInstructions++] = &tpl.opinfo;
-
-  initialized = true;
+  return table;
+}
+();
+
+Interpreter::Instruction Interpreter::GetInterpreterOp(UGeckoInstruction inst)
+{
+  // Check for the appropriate subtable ahead of time.
+  // (This is used by the cached interpreter and JIT, and called once per instruction, so spending a
+  // bit of extra time to optimise is worthwhile)
+  Interpreter::Instruction result = s_interpreter_op_table[inst.OPCD];
+  if (result == Interpreter::RunTable4)
+    return s_interpreter_op_table4[inst.SUBOP10];
+  else if (result == Interpreter::RunTable19)
+    return s_interpreter_op_table19[inst.SUBOP10];
+  else if (result == Interpreter::RunTable31)
+    return s_interpreter_op_table31[inst.SUBOP10];
+  else if (result == Interpreter::RunTable59)
+    return s_interpreter_op_table59[inst.SUBOP5];
+  else if (result == Interpreter::RunTable63)
+    return s_interpreter_op_table63[inst.SUBOP10];
+  else
+    return result;
+}
+
+void Interpreter::RunInterpreterOp(UGeckoInstruction inst)
+{
+  // Will handle subtables using RunTable4 etc.
+  s_interpreter_op_table[inst.OPCD](inst);
+}
+
+void Interpreter::RunTable4(UGeckoInstruction inst)
+{
+  s_interpreter_op_table4[inst.SUBOP10](inst);
+}
+void Interpreter::RunTable19(UGeckoInstruction inst)
+{
+  s_interpreter_op_table4[inst.SUBOP10](inst);
+}
+void Interpreter::RunTable31(UGeckoInstruction inst)
+{
+  s_interpreter_op_table4[inst.SUBOP10](inst);
+}
+void Interpreter::RunTable59(UGeckoInstruction inst)
+{
+  s_interpreter_op_table4[inst.SUBOP5](inst);
+}
+void Interpreter::RunTable63(UGeckoInstruction inst)
+{
+  s_interpreter_op_table4[inst.SUBOP10](inst);
 }
```

Comment:
Here's the source of the problem; I use `table4` for all of these by accident. The JITs are fine it seems. Although I'm not sure why this would be getting ran on the JITs; the whole splitting should mean that the JIT doesn't need to look at the interpreter at all now unless it's actually falling back.

Reply:
While the JIT doesn't run instructions that require a fallback all that often, there is a non-trivial number of instructions that require a fallback, so it sounds plausible to me that you would run into some of them at least once when playing a game.

Also, dvessel's stack trace was from while the apploader was running. We always use the interpreter for that.